ocean_lib/web3_internal/contract_base.py

```diff
@@ -199,16 +197,31 @@ def send_transaction(
         """
         contract_fn = getattr(self.contract.functions, fn_name)(*fn_args)
         contract_function = CustomContractFunction(contract_fn)
+
         _transact = {
             "from": ContractBase.to_checksum_address(from_wallet.address),
             "account_key": from_wallet.key,
             "chainId": self.web3.eth.chain_id,
-            "gasPrice": self.get_gas_price(self.web3),
         }
 
-        gas_price = os.environ.get(ENV_GAS_PRICE, None)
-        if gas_price:
-            _transact["gasPrice"] = gas_price
+        try:
+            history = self.web3.eth.fee_history(block_count=1, newest_block="latest")
+
+            _transact["gas"] = self.web3.eth.estimate_gas(_transact)
+            fee_tx = ContractBase.get_max_fee_per_gas(
+                web3=self.web3, tx=_transact, history=history
+            )
+            _transact.update(fee_tx)
+        except ValueError as e:
+            assert (
+                e.args[0]["message"] == "Method eth_feeHistory not supported."
+            ), f"Another error occurred: {e.args[0]}"
+
+            _transact["gasPrice"] = self.get_gas_price(self.web3)
```

Comment:
why are we calculating the gas price even if we have an env defined?

Reply:
I call the function get_gas_price so it includes the env defined if exists.